src/ray/core_worker/store_provider/plasma_store_provider.cc

```diff
@@ -307,8 +307,7 @@ Status CoreWorkerPlasmaStoreProvider::Get(
     should_break = timed_out || *got_exception;
 
     if ((previous_size - remaining.size()) < batch_ids.size()) {
-      unsuccessful_attempts++;
-      WarnIfAttemptedTooManyTimes(unsuccessful_attempts, remaining);
+      WarnIfFetchHanging(fetch_start_time_ms, remaining);
```

Comment:
Btw, just out of curiosity; how do we distinguish hanging vs task that creates the object is just not ready?

Reply:
Yeah, we shouldn't reach this line until the task has actually finished (this step is just about fetching the object from a remote node).